client.go

```diff
@@ -65,10 +53,18 @@ func (c *Client) ID() peer.ID {
 // Call performs an RPC call to a registered Server service and blocks until
 // completed. If dest is empty ("") or matches the Client's host ID, it will
 // attempt to use the local configured Server when possible.
-func (c *Client) Call(dest peer.ID, svcName string, svcMethod string, args interface{}, reply interface{}) error {
+func (c *Client) Call(dest peer.ID, svcName string, svcMethod string, args, reply interface{}) error {
+	ctx := context.Background()
+	return c.CallWithContext(ctx, dest, svcName, svcMethod, args, reply)
+}
+
+// CallWithContext performs a Call() with a user provided context. This gives
+// the user the possibility of cancelling the operation at any point.
+func (c *Client) CallWithContext(ctx context.Context, dest peer.ID, svcName, svcMethod string, args, reply interface{}) error {
```

Comment:
Following the example of [`database/sql`](https://golang.org/pkg/database/sql/#Stmt), could drop the 'With' in the function name. Also applies to `GoWithContext`.